internal/core/application/operator_service.go

```diff
@@ -961,57 +988,105 @@ func (o *operatorService) claimDeposit(
 	ctx context.Context,
 	infoPerAccount map[int]domain.AddressesInfo,
 	outpoints []TxOutpoint,
-) (domain.AddressesInfo, error) {
-	// first thing, retrieve all scripts  of the outpoints.
-	outpointScripts := make([]string, len(outpoints), len(outpoints))
+	depositType int,
+) error {
+	// group all addresses info by script
+	infoByScript := make(map[string]domain.AddressInfo)
+	for _, info := range infoPerAccount {
+		for s, i := range groupAddressesInfoByScript(info) {
+			infoByScript[s] = i
+		}
+	}
+
+	// for each outpoint retrieve the raw tx and output. If the output script
+	// exists in infoByScript, increment the counter of the related account and
+	// unblind the raw confidential output.
+	// Since all outpoints MUST be funds of the same account, at the end of the
+	// loop there MUST be only one counter matching the length of the give
+	// outpoints.
+	counter := make(map[int]int)
+	unspents := make([]domain.Unspent, len(outpoints), len(outpoints))
 	for i, v := range outpoints {
 		confirmed, err := o.explorerSvc.IsTransactionConfirmed(v.Hash)
 		if err != nil {
-			return nil, err
+			return err
 		}
 		if !confirmed {
-			return nil, ErrTxNotConfirmed
+			return ErrTxNotConfirmed
 		}
 
 		// TODO: Add a GetTransaction method to explorer interface to prevent
 		// direct usage of go-elements here.
 		txHex, err := o.explorerSvc.GetTransactionHex(v.Hash)
 		if err != nil {
-			return nil, err
+			return err
 		}
 		tx, _ := transaction.NewTxFromHex(txHex)
 
 		if len(tx.Outputs) <= v.Index {
-			return nil, ErrInvalidOutpoint
+			return ErrInvalidOutpoint
 		}
 
-		outpointScripts[i] = hex.EncodeToString(tx.Outputs[v.Index].Script)
-	}
+		txOut := tx.Outputs[v.Index]
+		script := hex.EncodeToString(txOut.Script)
+		if info, ok := infoByScript[script]; ok {
+			counter[info.AccountIndex]++
 
-	// By lopping on the map of the account provided, we search for the one
-	// whose scripts own ALL the given outpoints.
-	for _, info := range infoPerAccount {
-		accountScripts := make(map[string]struct{})
-		for _, in := range info {
-			accountScripts[in.Script] = struct{}{}
-		}
+			unconfidential, ok := transactionutil.UnblindOutput(
+				txOut,
+				info.BlindingKey,
+			)
+			if !ok {
+				return errors.New("unable to unblind output")
+			}
 
-		count := 0
-		for _, s := range outpointScripts {
-			if _, ok := accountScripts[s]; ok {
-				count++
+			unspents[i] = domain.Unspent{
+				TxID:            tx.TxHash().String(),
+				VOut:            uint32(v.Index),
+				Value:           unconfidential.Value,
+				AssetHash:       unconfidential.AssetHash,
+				ValueCommitment: bufferutil.CommitmentFromBytes(txOut.Value),
+				AssetCommitment: bufferutil.CommitmentFromBytes(txOut.Value),
```

Comment:
should be txOut.Asset?

Reply:
yeap, this is the issue with 'Invalid prefix issue' happening cause wrong value for AssetCommitment.

Reply:
fixing it, thanks!